src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java

```diff
@@ -120,6 +117,11 @@ private Object[] invocationArgs(DataFetchingEnvironment environment, ProcessingE
 
     private Object buildArg(Type p, GraphQLType graphQLType, Object arg) {
         if (arg == null) {
+            // for Optional parameters null should be returned as Optional.empty() to show a request for a null value
+            // and not including the parameter in the query at all should be returned as null to show "undefined" value / not set
+            if ((p instanceof ParameterizedType && ((ParameterizedType) p).getRawType() == Optional.class)) {
```

Comment:
so, it's assumed lines 141-147 will be executed before this in the recursion? and thus adding the "undefined" `null`s there?